### PR TITLE
[Merged by Bors] - feat: the homological functor on the homotopy category of cochain complexes in an abelian category

### PR DESCRIPTION
**Pull request by joelriou**
In this PR, it is shown that the homology functors on the homotopy category of cochain complexes in an abelian category are homological functors (in the sense that they send distinguished triangles to exact sequences). This PR also shows that the category of homological complexes in an abelian category is an abelian category.

---

- [x] depends on: #11740
- [x] depends on: #11759

Note:  the two new definitions like `Abelian.isColimitMapCoconeOfCokernelCoforkOfπKernelConditionOfEpi` are also used in order to show that the category of presheaves of modules is abelian #12721.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Abelian/Basic.lean [hunk 1]
```diff
@@ -498,6 +499,44 @@ theorem monoLift_comp [Mono f] {T : C} (g : T ⟶ Y) (hg : g ≫ cokernel.π f =
...
 #align category_theory.abelian.mono_lift_comp CategoryTheory.Abelian.monoLift_comp
 
+section
+
+variable {D : Type*} [Category D] [HasZeroMorphisms D]
+
+/-- If `F : D ⥤ C` is a functor to an abelian category, `i : X ⟶ Y` is a morphism
+admitting a cokernel such that `F` preserves this cokernel and `F.map i` is a mono,
+then `F.map X` identifies to the kernel of `F.map (cokernel.π i)`. -/
+noncomputable def isLimitMapConeOfKernelForkOfιCokernelConditionOfMono
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Abelian/Basic.lean:509]: These names are intimidatingly long... would `isLimitMapConeOfKernelForkOfι` be specific enough?

**Review by jcommelin — APPROVED** [review]
LGTM, but one question about names.

File: Mathlib/CategoryTheory/Abelian/Basic.lean (hunk 1 above)
Comment (joelriou) [reply Mathlib/CategoryTheory/Abelian/Basic.lean]: Thanks for the suggestions! I have shortened these names.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**